common/src/AABBTree.h

```diff
@@ -208,97 +175,119 @@ class AABBTree : public NodeTree<T,S,U,Cmp> {
         m_height(0) {
             assert(m_left != nullptr);
             assert(m_right != nullptr);
+
+            m_left->m_parent = this;
+            m_right->m_parent = this;
+
+            updateHeight();
+        }
+
+    private: // node removal private
+        /**
+         * Children (or grandchildren etc.) changed. Update the height and bounds
```

Comment:
Add a note that this will recurse along the parent chain until it reaches the root.